### ISSUE TITLE
Certain older schemas do not render due to definition error

### ISSUE THREAD

**Issue by helen-m-lin** [AllenNeuralDynamics/aind-metadata-entry-js#131]
**Describe the bug**
Certain older schemas do not render and cause app to crash with errors.
Schema versions that cause errors:
- mri_session 0.2.0 - 0.2.7: `Error: Could not find a definition for #/$defs/Abcam.`
- rig 0.1.0 - 0.1.7: `Error: Could not find a definition for #/$defs/SizeUnit.`

**To Reproduce**
1. Go to https://metadata-entry.allenneuraldynamics.org/
2. Click on 'Select schema'
3. Select any schema impacted as described above (e.g. mri_session 0.2.6)
4. See app has crashed and observe error in console.

**Expected behavior**
The selected schema and version should render.

**Screenshots**
![image](https://github.com/AllenNeuralDynamics/aind-metadata-entry-js/assets/46795546/e88f6288-60ac-4c93-b2bd-792470a792a0)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome, Edge
 - Version: App version 0.1.6

**Additional context**

**Comment by jtyoung84** [comment]
1 - Catch error and display message
2 - If the schemas are really old, we can consider adding a filter to filter out versions less than a number
3 - Try to understand where this error is coming from

**Comment by helen-m-lin** [comment]
I tested all schemas/versions and found the following errors causing app crashes:
- mri_session 0.2.9, 0.2.8: `Error: Could not find a definition for #/$defs/Neurophotometrics`
- mri_session 0.2.7-0.2.0: `Error: Could not find a definition for #/$defs/Abcam.`
- rig 0.1.0-0.1.7: `Error: Could not find a definition for #/$defs/SizeUnit.`

**Cause:** Unhandled errors were caused by incorrect state handling in the `RenderForm` component.
- React tries to re-use as much of the existing DOM as possible when the state is updated.
- In this case, when a new schema/version is selected in the parent component, React tries to re-use parts of the existing form which causes certain schemas to crash.
- **Mitigation in PR #142**
